install_python_ubuntu.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+# Copyright (c) 2023 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# This script is used inside the Dockerfile.ubuntu. 
+# Modify this file, if you want to build OVMS with non-default version of Python.
+# See available versions on https://www.python.org/ftp/python/
+# Default version is Python 3.10.12
+
+PYTHON3_VERSION_MAJOR="10"
+PYTHON3_VERSION_MINOR="12"
+
+wget https://www.python.org/ftp/python/3.${PYTHON3_VERSION_MAJOR}.${PYTHON3_VERSION_MINOR}/Python-3.${PYTHON3_VERSION_MAJOR}.${PYTHON3_VERSION_MINOR}.tgz 
+tar -xzf Python-3.${PYTHON3_VERSION_MAJOR}.${PYTHON3_VERSION_MINOR}.tgz 
+cd Python-3.${PYTHON3_VERSION_MAJOR}.${PYTHON3_VERSION_MINOR} 
+./configure --build=x86_64-linux-gnu \
+        --enable-loadable-sqlite-extensions \
+		--enable-optimizations \
+		--enable-option-checking=fatal \
+		--enable-shared \
+		--with-lto \
+		--with-system-expat
+make -j ${nproc} 
+make install 
+ldconfig
+cd .. && rm -rf Python-3.${PYTHON3_VERSION_MAJOR}.${PYTHON3_VERSION_MINOR}.tgz Python-3.${PYTHON3_VERSION_MAJOR}.${PYTHON3_VERSION_MINOR}
```

Comment:
If any error occures previoulsy, we will not delete tar.gz nor Python3. dir.

Please take a peak into this article:

https://stackoverflow.com/questions/15656492/writing-try-catch-finally-in-shell